### PR TITLE
Add SVG files sniff for PHP openings

### PR DESCRIPTION
**Pull request by gudmdharalds**
Look for <?php and <?= in SVG files, and give a warning when that happens.

### CODE REVIEW

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	 *                                               the token was found.
+	 *
+	 * @return void
+	 */
+	public function process( File $phpcsFile, $stackPtr ) {
+		// Get tokens.
+		$tokens = $phpcsFile->getTokens();
+
+		$nxt = $phpcsFile->findNext(
+			T_OPEN_TAG,
```
Comment (david-binda) [inline WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: the `process` method is entered only when `T_OPEN_TAG` is encountered. That say, mere entering of this method makes it explicit that an open tag has been found, and there is no need to check whether it really is an open tag, or of what type.

Thus, in case we are in this method, we can directly return the Warning.

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	/**
+	 * Returns the token types that this sniff is interested in.
+	 *
+	 * We want everything function-related.
+	 *
+	 * @return array(int)
+	 */
+	public function register() {
+		return array(
+			T_OPEN_TAG,
```
Comment (david-binda) [inline WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: There is also `T_OPEN_TAG_WITH_ECHO` checking for `<?=` which should be part of this sniff, imho.

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		);
+
+		$tokens[ $nxt ]['content'] =
+			strtolower( $tokens[ $nxt ]['content'] );
+
+		$found1 =
+			strpos( $tokens[ $nxt ]['content'], '<?php' );
+
+		$found2 =
+			strpos( $tokens[ $nxt ]['content'], '<?' );
```
Comment (david-binda) [inline WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: Unless short open tags are allowed in the PHP config on the server where this runs, those won't be catched by `T_OPEN_TAG`.

Only way to catch short open tags is to have short open tags allowed on the server. But it only makes sense in case it mirrors the production config - I believe that short open tags are disallowed on both, WPCOM and VIP GO now.

Here's a related issue in PHPCS covering the topic of short open tags and it's catching: https://github.com/squizlabs/PHP_CodeSniffer/issues/1571 and here's a direct link to how it's handled in the Generic.PHP.DisallowShortOpenTag sniff: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Generic/Sniffs/PHP/DisallowShortOpenTagSniff.php#L32,L34

File: WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	/**
+	 * Returns the lines where warnings should occur.
+	 *
+	 * @return array <int line number> => <int number of warnings>
+	 */
+	public function getWarningList() {
+		return array(
+			7 => 1,
```
Comment (GaryJones) [inline WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.php]: Where's the test file that this relates to? I don't see it in this PR.

Comment (gudmdharalds) [reply WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.php]: Has been added now. :-)

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php (hunk 3 above)
Comment (gudmdharalds) [reply WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: I think we will let it suffice to look for `<?php` and `<?=` because of how the configuration is on our servers.

Comment (gudmdharalds) [reply WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: And, to add: this test will be in conjunction with the `Generic.PHP.DisallowShortOpenTag` sniff.

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	 * @param int                         $stackPtr  The position in the stack where
+	 *                                               the token was found.
+	 *
+	 * @return void
+	 */
+	public function process( File $phpcsFile, $stackPtr ) {
+		$tokens = $phpcsFile->getTokens();
+
+		// Make sure it is a SVG file.
+		$file_extension = strtolower(
```
Comment (GaryJones) [inline WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: This `strtolower( ... )` could be extracted into its own method to make the code more self-documenting. Line 62 would then be something like:

```php
if ( 'svg' !== $this->get_file_extension( $phpcsFile->path ) ) {
```

File: WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.svg [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<svg width="299px" height="31px" viewBox="0 0 299 11" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+	<!-- Generator: Tezt -->
+	<title>Page 100</title>
+	<desc>Created with Sketch.</desc>
+
+
+
+<?php
```
Comment (GaryJones) [inline WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.svg]: I'd like to see some single-line opening and closing tags, and with something in between them, to ensure that the logic handles some likely uses. Just having the opening and closing tags (both types), nicely at the start of different lines, with nothing in between, isn't pushing to find all the edge cases.

Comment (gudmdharalds) [reply WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.svg]: Thanks! I've added this now :-)

File: WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php (hunk 5 above)
Comment (gudmdharalds) [reply WordPressVIPMinimum/Sniffs/SVG/HTMLCodeSniff.php]: Done :-)

File: WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.svg (hunk 6 above)
Comment (GaryJones) [reply WordPressVIPMinimum/Tests/SVG/HTMLCodeSniff.svg]: Sorry - I wasn't clear - I wanted to see multiple efforts with and without extra content:

```
<?php
?><?php echo 'foo'; ?><?= ?><?php ?><?=

?>
```

etc.
